coin_prices_from_coingecko/output.go

```diff
@@ -39,6 +27,15 @@ func WriteOutput(output any) uint64 {
 }
 
 func WriteError(err error) uint64 {
-	data := Result{}.JSONMarshalWithError(err)
+	result := Result{
+		Success: false,
+		Error:   err.Error(),
+		Value:   nil,
+	}
+	data, marshalErr := json.Marshal(result)
+	if marshalErr != nil {
+		basm.Log(fmt.Sprintf("Error marshalling Result: %v", marshalErr))
+		data = []byte(`{ "Success": false, "Error": "failed to marshal result" , "Value": null }`)
```

Comment:
Updated `WriteError` to follow the `WriteOutput` approach. That is, we define the result struct and try to json marshal it. Should that fail, however, we return this hardcoded result string